### PR TITLE
`azurerm_redis_enterprise_cluster` - upgrade api version and support new skus

### PR DESCRIPTION
**Pull request by sinbai**
As requested by the service team, the `Enterprise_E5`, `Enterprise_E200` and `Enterprise_E400` SKUs of resource `azurerm_redis_enterprise_cluster` are supported.

Notes: 

- Given supporting new SKUs is just a simple updates, it is merged with the upgrade API into one PR.

- Although this is a preview release, the service team has promised that there won't be any breaking changes.

- Due to the comments in https://github.com/hashicorp/terraform-provider-azurerm/pull/23185, this change does not have test cases but is tested locally.

### CODE REVIEW

File: internal/services/redisenterprise/validate/redis_enterprise_cluster_sku_name.go [hunk 1]
```diff
@@ -71,11 +71,14 @@ func RedisEnterpriseClusterSkuName(v interface{}, k string) (warnings []string,
 
 func getValidRedisEnterpriseClusterSkus() []string {
```
Comment (tombuildsstuff) [inline internal/services/redisenterprise/validate/redis_enterprise_cluster_sku_name.go:72]: FWIW this function can be replaced by `redisenterprise.PossibleValuesForSkuName()` which will give us all the possible values for the `SkuName` constant (which is the list below) - meaning we'll get these automatically in future SDK upgrades

**Review by tombuildsstuff — APPROVED** [review]
One comment about reducing duplication (since `hashicorp/go-azure-sdk` already exposes a method we can use) - but if we can fix that up this otherwise LGTM 👍

File: internal/services/redisenterprise/validate/redis_enterprise_cluster_sku_name.go (hunk 1 above)
Comment (sinbai) [reply internal/services/redisenterprise/validate/redis_enterprise_cluster_sku_name.go]: Thanks for your feedback. The code has been updated. Could you please take another look?

**Review by stephybun — APPROVED** [review]
Thanks @sinbai LGTM 👍

**Pull request merged**